README.md

```diff
@@ -0,0 +1,63 @@
+# GA4GH SDK
+
+## Synopsis
+
+A Generic SDK and CLI for GA4GH API services
+
+## Basic usage
+
+## Installation
+
+## Building
+
+First, clone the repository, and then run the following command to automatically generate models using OpenAPI specifications:
+```
+bash ./build-models.sh
+```
+
+To build the project:
+```
+cargo build
+```
+
+## Running the tests
+
+Before running the tests, you need to install Funnel, a task execution system that is compatible with the GA4GH TES API. Follow the instructions in the Funnel Developer's Guide to install Funnel: https://ohsu-comp-bio.github.io/funnel/docs/development/developers/. Note that the link may change over time, so refer to the official Funnel repository or website for the most up-to-date information.
+
+Once you have installed Funnel, you can run the tests. This will automatically run Funnel as well:
+
+```
+bash ./run-tests.sh
+```
+or, you can run using:
+```
+cargo nextest run
+```
+For checking the unit coverage, you can run:
+```
+cargo llvm-cov nextest
+```
+
+To test the CI/CD workflow locally, install `act` and run the following command:
+```
+act -j build --container-architecture linux/amd64 -P ubuntu-latest=ubuntu:24.04 --reuse
+# Note: The specified Ubuntu version (24.04) may change in the future. 
+# Please check for the latest version when running this command.
+```
```

Comment:
Ideally, you'd add a basic `README.md` skeleton as one of your first (if not the first) PRs - then update the docs when new info becomes available, i.e., on installation, building, running tests etc.

So you could consider removing this information (along with the build info) and have this merged first. Then add additional docs continuously witch each PR, where applicable (e.g., when tests are defined, the CI workflow is added etc.).

However, hoping that everything will come together just right after merging the individual PRs (after all, they are result of breaking apart a big PR that was, presumably, already working), I am fine with keeping things as they are now if you can live with the limited chronology of the commit history.

In the future though, I think we should adhere to continuous testing, documentation etc.

Reply:
Ok, I will keep this in mind for any future projects I do. Also, I think that this file is going to be updated a lot in the coming PRs, and I think that this is just the basic version..
I think I will just mark this conversation as resolved, so that this PR can get merged